src/service/performance-impl.js

```diff
@@ -715,30 +715,27 @@ export class Performance {
    *     this directly.
    */
   tick(label, opt_delta) {
-    const value = opt_delta == undefined ? this.win.Date.now() : undefined;
+    let data;
+    let storedVal;
+
+    // Absolute value case (not delta).
+    if (opt_delta == undefined) {
+      // Marking only makes sense for non-deltas.
+      this.mark(label);
+      const now = this.win.Date.now();
+      data = dict({'label': label, 'value': now});
+      storedVal = now - this.initTime_;
+    } else {
+      data = dict({'label': label, 'delta': Math.max(opt_delta, 0)});
+      storedVal = Math.max(opt_delta, 0);
+    }
 
-    const data = dict({
-      'label': label,
-      'value': value,
-      // Delta can negative, but will always be changed to 0.
-      'delta': opt_delta != null ? Math.max(opt_delta, 0) : undefined,
-    });
     if (this.isMessagingReady_ && this.isPerformanceTrackingOn_) {
       this.viewer_.sendMessage('tick', data);
     } else {
       this.queueTick_(data);
     }
-    // Mark the event on the browser timeline, but only if there was
-    // no delta (in which case it would not make sense).
-    if (arguments.length == 1) {
-      this.mark(label);
-    }
 
-    // Store certain page visibility metrics to be exposed as analytics
-    // variables.
-    const storedVal = Math.round(
-      opt_delta != null ? Math.max(opt_delta, 0) : value - this.initTime_
-    );
```

Comment:
I actually find this less readable. I think there's a ToTT for it somewhere but I also found this: https://refactoring.com/catalog/reduceScopeOfVariable.html

Reply:
You prefer the `opt_delta` condition repeated 3 times?

Reply:
Speaking of this line, I have a question. Why in the absolute case (relative to epoch) do we set `storedVal` to the delta?

Reply:
answer: its used in the url-rewriter, and non-viewer use cases care more about relative than absolute times.